pyro/infer/autoguide/guides.py

```diff
@@ -656,44 +657,75 @@ def _loc_scale(self, *args, **kwargs):
         return self.loc, scale
 
 
-class AutoIAFNormal(AutoContinuous):
+class AutoNormalizingFlow(AutoContinuous):
     """
     This implementation of :class:`AutoContinuous` uses a Diagonal Normal
-    distribution transformed via a :class:`~pyro.distributions.transforms.AffineAutoregressive`
-    to construct a guide over the entire latent space. The guide does not depend on the model's
-    ``*args, **kwargs``.
+    distribution transformed via one of the normalizing flows in
+    :mod:`~pyro.distributions.transforms`to construct a guide over the entire
+    latent space. The guide does not depend on the model's ``*args, **kwargs``.
 
     Usage::
 
-        guide = AutoIAFNormal(model, hidden_dim=latent_dim)
+        guide = AutoNormalizingFlow(model, 2, affine_autoregressive)
         svi = SVI(model, guide, ...)
 
     :param callable model: a generative model
-    :param int hidden_dim: number of hidden dimensions in the IAF
+    :param num_flows: number of normalizing flows
+    :param init_transform_fn: a callable which when provided with the latent
+        dimension and additional keyword arguments (`init_transform_kwargs`)
+        returns an instance of
+        :class:`~pyro.distributions.torch_tranform.TransformModule`.
     :param callable init_loc_fn: A per-site initialization function.
         See :ref:`autoguide-initialization` section for available functions.
+    :param init_tranform_kwargs: other keyword arguments taken by
+        `init_transform_fn`.
     """
 
-    def __init__(self, model, hidden_dim=None, init_loc_fn=init_to_median):
-        self.hidden_dim = hidden_dim
-        self.arn = None
+    def __init__(self, model, num_flows, init_transform_fn, init_loc_fn=init_to_median,
+                 **init_transform_kwargs):
+        self.num_flows = num_flows
+        self.transform = None
+        self._init_transform_fn = init_transform_fn
+        self._init_transform_kwargs = init_transform_kwargs
         super().__init__(model, init_loc_fn=init_loc_fn)
 
     def get_posterior(self, *args, **kwargs):
         """
         Returns a diagonal Normal posterior distribution transformed by
-        :class:`~pyro.distributions.transforms.iaf.InverseAutoregressiveFlow`.
+        :class:`~pyro.distributions.torch_tranform.TransformModule`.
         """
         if self.latent_dim == 1:
             raise ValueError('latent dim = 1. Consider using AutoDiagonalNormal instead')
```

Comment:
i'm not sure if this `ValueError` is still needed? @stefanwebb are any of the current flows non-trivial in `D=1`?

Reply:
Not yet, but `SplineCoupling` and some other ones will be (to be pushed shortly). So I recommend removing this assertion

Reply:
Okay, removed this.